rotkehlchen/db/history_events.py

```diff
@@ -236,23 +235,28 @@ def delete_events_by_location(
     ) -> None:
         """Delete all relevant non-customized events for a given location
 
-        Also set evm_tx_mapping as non decoded so they can be redecoded later
+        Also set evm_tx_mapping as non decoded so they can be re-decoded later
         """
-        customized_event_ids = self.get_customized_event_identifiers(cursor=write_cursor, location=location)  # noqa: E501
-        whereclause = 'WHERE location=?'
-        if (length := len(customized_event_ids)) != 0:
-            whereclause += f' AND history_events.identifier NOT IN ({", ".join(["?"] * length)})'
-            bindings = [location.serialize_for_db(), *customized_event_ids]
-        else:
-            bindings = (location.serialize_for_db(),)  # type: ignore  # different type of elements in the list
-
-        transaction_hashes = write_cursor.execute(f'SELECT evm_events_info.tx_hash FROM history_events INNER JOIN evm_events_info ON history_events.identifier=evm_events_info.identifier {whereclause}', bindings).fetchall()  # noqa: E501
-        write_cursor.execute(f'DELETE FROM history_events {whereclause}', bindings)
-
-        if location != Location.ZKSYNC_LITE and len(transaction_hashes) != 0:
-            write_cursor.executemany(
-                'DELETE from evm_tx_mappings WHERE tx_id IN (SELECT identifier FROM evm_transactions WHERE tx_hash=? AND chain_id=?) AND value=?',  # noqa: E501
-                [(x[0], location.to_chain_id(), EVMTX_DECODED) for x in transaction_hashes],
+        customized_events = write_cursor.execute(
+            'SELECT COUNT(*) FROM history_events_mappings WHERE name=? AND value=?',
+            (HISTORY_MAPPING_KEY_STATE, HISTORY_MAPPING_STATE_CUSTOMIZED),
+        ).fetchone()[0]
+        querystr = (
+            'DELETE FROM history_events WHERE identifier IN ('
+            'SELECT H.identifier from history_events H INNER JOIN evm_events_info E '
+            'ON H.identifier=E.identifier AND E.tx_hash IN '
+            '(SELECT tx_hash FROM evm_transactions) AND H.location = ?)'
+        )
+        bindings: tuple = (location.serialize_for_db(),)
+        if customized_events != 0:
+            querystr += ' AND identifier NOT IN (SELECT parent_identifier FROM history_events_mappings WHERE name=? AND value=?)'  # noqa: E501
+            bindings += (HISTORY_MAPPING_KEY_STATE, HISTORY_MAPPING_STATE_CUSTOMIZED)
+
+        if location != Location.ZKSYNC_LITE:
```

Comment:
I don't get this. Why is this here? Soif the location = Location.ZKSYNC_LITE what does this function do? If it does nothing then you should change the type of the function and make sure it never happens.